subsys/bluetooth/mesh/vnd/silvair_enocean_srv.c

```diff
@@ -73,11 +116,15 @@ static void state_machine_button_process(
 			K_MSEC(CONFIG_BT_MESH_SILVAIR_ENOCEAN_WAIT_TIME));
 		break;
 	case BT_ENOCEAN_BUTTON_RELEASE:
-		if (button->state !=
-				BT_MESH_SILVAIR_ENOCEAN_STATE_DELTA ||
-				button->target != on_off) {
+		if (button->state != BT_MESH_SILVAIR_ENOCEAN_STATE_DELTA ||
+		    !k_work_delayable_is_pending(&button->timer) ||
```

Comment:
Will this k_work_delayable check put us back where we were, so that this now makes it turn off again if you hold the button, dimming up until it times out, and then release it? Or am I misunderstanding the purpose of this check?

Reply:
Yes, it will. This is what spec mandates when dimming times out:
> 3.3.3.3 Delta Timeout Behavior
When a button is pressed for a long time (over 6 seconds), the Generic Delta Set Unacknowledged
messages are not sent anymore.
When the button is released, the Generic OnOff Set Unacknowledged is sent.